aiida/manage/configuration/profile.py

```diff
@@ -92,33 +77,20 @@ def defaults(cls):  # pylint: disable=no-self-use,no-self-argument
             }
         }
 
-    @classmethod
-    def contains_unknown_keys(cls, dictionary):
-        """Return whether the profile dictionary contains any unsupported keys.
-
-        :param dictionary: a profile dictionary
-        :return: boolean, True when the dictionay contains unsupported keys
-        """
-        return set(dictionary.keys()) - set(cls._map_config_to_internal.keys())
-
-    def __init__(self, name, attributes, from_config=False):
-        if not isinstance(attributes, collections.abc.Mapping):
-            raise TypeError(f'attributes should be a mapping but is {type(attributes)}')
+    def __init__(self, name: str, config: Mapping[str, Any], validate=False):
+        """Load a profile with the profile configuration."""
+        if not isinstance(config, collections.abc.Mapping):
+            raise TypeError(f'config should be a mapping but is {type(config)}')
+        # to-do currently this is not actually used anywhere,
+        # because e.g. the documentation is loaded with an incomplete (dummy) configuration
+        # in actual usage though, this could lead to later key errors, when retrieving an attribute
+        if validate and not set(config.keys()).issuperset(self.REQUIRED_KEYS):
```

Comment:
Where is this excepting then if `validate` were to be `True`? If it is not needed, shouldn't we just get rid of `validate`?

Reply:
As it says in the `to-do` comment above, I think this should be on always, since otherwise you could have obscure failures, where you get `KeyError` attribute retrieval on e.g. `Profile.storage_backend`. The reason why it is not at present, is that in some testing fixtures and the `load_documentation` function, an incomplete config is supplied to `Profile`, so this would fail. We should maybe just add "dummy" config to them (for e.g. rabbitmq config), so we can always run this validation.

Reply:
I read the to-do comment but I don't understand it. If you don't currently use the `validate` keyword anywhere, then why not just remove it. You say it is needed for the `load_documentation` one, but that means it _is_ being used and so the `to-do` is incorrect and can be removed?

Reply:
In ba998ffbc76785c1b5209089dc6ce42c2aa29bc9, I have "fixed" places that were loading incomplete config, and turned on validation by default.

Reply:
Cheers, I think this still needs to be revisited at some point, but we can do that later.